builtInFunctions/esdtMetaDataRecreate.go

```diff
@@ -122,10 +122,10 @@ func checkUpdateArguments(
 }
 
 type esdtStorageInfo struct {
-	esdtData     *esdt.ESDigitalToken
-	esdtTokenKey []byte
-	nonce        uint64
-	isDynamic    bool
+	esdtData            *esdt.ESDigitalToken
+	esdtTokenKey        []byte
+	nonce               uint64
+	metaDataInSystemAcc bool
 }
 
 func getEsdtInfo(
```

Comment:
is this function used only by the new functions ?

If yes, than it is correct. If it is used by old functions than you should still read SFT/MetaESDT data from system account.

Reply:
This is used only on the new functions.